fate/src/main/java/org/apache/accumulo/fate/util/Retry.java

```diff
@@ -0,0 +1,356 @@
+  }
+
+  public interface NeedsRetries {
+    /**
+     * @return this builder with the maximum number of retries set to unlimited
+     */
+    NeedsRetryDelay infiniteRetries();
+
+    /**
+     * @param max
+     *          the maximum number of retries to set
+     * @return this builder with the maximum number of retries set to the provided value
+     */
+    NeedsRetryDelay maxRetries(long max);
+  }
+
+  public interface NeedsRetryDelay {
+    /**
+     * @param duration
+     *          the amount of time to wait before the first retry; input is converted to milliseconds, rounded down to the nearest
+     * @return this builder with the initial wait period set
+     */
+    NeedsTimeIncrement retryAfter(long duration, TimeUnit unit);
+  }
+
+  public interface NeedsTimeIncrement {
+    /**
+     * @param duration
+     *          the amount of additional time to add before each subsequent retry; input is converted to milliseconds, rounded down to the nearest
+     * @return this builder with the increment amount set
+     */
+    NeedsMaxWait incrementBy(long duration, TimeUnit unit);
+  }
+
+  public interface NeedsMaxWait {
+    /**
+     * @param duration
+     *          the maximum amount of time to which the waiting period between retries can be incremented; input is converted to milliseconds, rounded down to
+     *          the nearest
+     * @return this builder with a maximum time limit set
+     */
+    NeedsLogInterval maxWait(long duration, TimeUnit unit);
+  }
+
+  public interface NeedsLogInterval {
+    /**
+     * @param duration
+     *          the minimum time interval between logging that a retry is occurring; input is converted to milliseconds, rounded down to the nearest
+     * @return this builder with a logging interval set
+     */
+    BuilderDone logInterval(long duration, TimeUnit unit);
+  }
+
+  public interface BuilderDone {
+    /**
+     * Create a RetryFactory from this builder which can be used to create many Retry objects with the same settings.
+     *
+     * @return this builder as a factory; intermediate references to this builder cannot be used to change options after this has been called
+     */
+    RetryFactory createFactory();
+
+    /**
+     * Create a single Retry object with the currently configured builder settings.
+     *
+     * @return a retry object from this builder's settings
+     */
+    Retry createRetry();
+  }
+
+  public interface RetryFactory {
+    /**
+     * Create a single Retry object from this factory's settings.
+     *
+     * @return a retry object from this factory's settings
+     */
+    Retry createRetry();
+  }
+
+  public static NeedsRetries builder() {
+    return new RetryFactoryBuilder();
+  }
+
+  private static class RetryFactoryBuilder implements NeedsRetries, NeedsRetryDelay, NeedsTimeIncrement, NeedsMaxWait, NeedsLogInterval, BuilderDone,
```

Comment:
Since you changed to the builder pattern why not rename to "RetryBuilder".  I think its confusing to intermingle the terms.

Reply:
Technically... it's a builder for RetryFactory *AND* Retry.

Reply:
Seems like overkill since we only use the default settings.

Reply:
Yeah I understand why you did it that way, it just seems overly complex for internal code.  I was thinking we could just modify the factory constructor to not have any parameters and static methods to change something other than the defaults.

Reply:
I was trying to move away from "defaults", because every retry scenario is different, and code in Accumulo tends to be written with the bad habit of using what's there, rather than thinking through what makes sense in the specific context.